woocommerce/woocommerce
# Make the price filter widget tax aware redux

Supercedes #21942
Closes #21203

#21942 semi-solved the problem but relied on some extra inputs to see which slider was used. We have price slider query dedicated methods so we can handle the tax subtraction there instead.

This PR:
- Moves the logic from #21942 to a more appropriate location
- Implements https://github.com/woocommerce/woocommerce/pull/22533
- Uses standard tax class only for the inc tax calculation, with a filter if needed. We cannot guess which tax class to use so standard makes the most sense.
- Implements a step of 10 so the ranges increment +10 at a time. This will produce more cache hits plus makes it slightly more user friendly as it can be hard to get a specific price with +1 increments.

**How to test the changes in this Pull Request:**

1. Enable taxes on your store and set a tax rate
2. Choose to enter product prices excl tax
3. Choose to display product prices incl tax on the frontend.
4. Add a Price filter widget and view the page with the widget.
5. Ensure that the filter range matches the range of the prices incl tax in your catalog.
6. Do a filter search with a price range and check that it returns the correct products.

> Fix - Make price filter widget tax aware when you have excl prices entered but show prices incl of tax.
